internal/receiver/awscontainerinsightreceiver/cadvisor/cadvisor_linux.go

```diff
@@ -0,0 +1,224 @@
+// +build linux
+
+package cadvisor
+
+import (
+	"errors"
+	"flag"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/cadvisor/cache/memory"
+	cadvisormetrics "github.com/google/cadvisor/container"
+	"github.com/google/cadvisor/container/containerd"
+	"github.com/google/cadvisor/container/crio"
+	"github.com/google/cadvisor/container/docker"
+
+	common "github.com/aws-observability/aws-otel-collector/internal/aws/containerinsightcommon"
+	"github.com/aws-observability/aws-otel-collector/internal/receiver/awscontainerinsightreceiver/cadvisor/extractors"
+	"github.com/aws-observability/aws-otel-collector/internal/receiver/awscontainerinsightreceiver/host"
+	"go.uber.org/zap"
+
+	"github.com/google/cadvisor/container/systemd"
+	cinfo "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/utils/sysfs"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// The amount of time for which to keep stats in memory.
+const statsCacheDuration = 2 * time.Minute
+
+// Max collection interval, it is not meaningful if allowDynamicHousekeeping = false
+var maxHousekeepingInterval = 15 * time.Second
+
+// When allowDynamicHousekeeping is true, the collection interval is floating between 1s(default) to maxHousekeepingInterval
+var allowDynamicHousekeeping = true
+
+const defaultHousekeepingInterval = 10 * time.Second
+
+type Cadvisor struct {
+	logger                *zap.Logger
+	nodeName              string //get the value from downward API
+	manager               manager.Manager
+	ebsVolume             *host.EbsVolume
+	version               string
+	machineInfo           *host.MachineInfo
+	containerOrchestrator string
+}
+
+func overrideCadvisorFlagDefault(logger *zap.Logger) {
+	flagOverrides := map[string]string{
+		// Override the default cAdvisor housekeeping interval.
+		"housekeeping_interval": defaultHousekeepingInterval.String(),
+		// override other defaults (in future)
+	}
+	for name, defaultValue := range flagOverrides {
+		if f := flag.Lookup(name); f != nil {
+			f.DefValue = defaultValue
+			f.Value.Set(defaultValue)
+		} else {
+			logger.Warn("Expected cAdvisor flag not found", zap.String("name", name))
```

Comment:
we dont' set `housekeeping_interval` in OTEL collector args. will we always have the warning msg here?

Reply:
This code is used to override some default setting for cadvisor because it doesn't provide an api. Some flag might not be there with new version of cadvisor (if we later upgrade cadvisor). So it makes sense to keep the check here.